ui-spacetimechart/src/utils/canvas.ts

```diff
@@ -273,3 +273,39 @@ export function drawPathExtremity(
     drawPathStopExtremity(ctx, timePixel, spacePixel, swapAxis);
   }
 }
+
+/**
+ *
+ * @param minT number timestamp
+ * @param maxT number timestamp
+ * @param timeRanges time frames (24h, 12h, 6h, …)
+ * @param gridlinesLevels width of the lines for each time frame
+ * @returns Record<number, number>
+ * Keys are times in ms
+ * Values are the highest level on each time
+ */
+
+export function displayElementsBasedOnZoom<T>(
```

Comment:
I wonder if we could find a better name of this function.

The function itself doesn't deal with the zoom so it sounds a bit weird to mention it there.

Maybe `computeMarksFromTimeLevels()`?

Reply:
`computeVisibleTimeMarkers` ?

Reply:
Works for me!